crates/iceberg/src/spec/values.rs

```diff
@@ -996,10 +1028,593 @@ mod timestamptz {
+                        }
+                    } else {
+                        return Err(Error::new(
+                            ErrorKind::DataInvalid,
+                            format!("Type {} should be a map", ty),
+                        ));
+                    }
+                }
+            };
+            Ok(raw)
+        }
+
+        pub fn try_into(self, ty: &Type) -> Result<Option<Literal>, Error> {
+            let invalid_err = |v: &str| {
+                Error::new(
+                    ErrorKind::DataInvalid,
+                    format!(
+                        "Unable to convert raw literal ({}) fail convert to type {} for: type mismatch",
+                        v, ty
+                    ),
+                )
+            };
+            let invalid_err_with_reason = |v: &str, reason: &str| {
+                Error::new(
+                    ErrorKind::DataInvalid,
+                    format!(
+                        "Unable to convert raw literal ({}) fail convert to type {} for: {}",
+                        v, ty, reason
+                    ),
+                )
+            };
+            match self {
+                RawLiteralEnum::Null => Ok(None),
+                RawLiteralEnum::Boolean(v) => Ok(Some(Literal::bool(v))),
+                RawLiteralEnum::Int(v) => match ty {
+                    Type::Primitive(PrimitiveType::Int) => Ok(Some(Literal::int(v))),
+                    Type::Primitive(PrimitiveType::Date) => Ok(Some(Literal::date(v))),
+                    _ => Err(invalid_err("int")),
+                },
+                RawLiteralEnum::Long(v) => match ty {
+                    Type::Primitive(PrimitiveType::Long) => Ok(Some(Literal::long(v))),
+                    Type::Primitive(PrimitiveType::Time) => Ok(Some(Literal::time(v))),
+                    Type::Primitive(PrimitiveType::Timestamp) => Ok(Some(Literal::timestamp(v))),
+                    Type::Primitive(PrimitiveType::Timestamptz) => {
+                        Ok(Some(Literal::timestamptz(v)))
+                    }
+                    _ => Err(invalid_err("long")),
+                },
+                RawLiteralEnum::Float(v) => match ty {
+                    Type::Primitive(PrimitiveType::Float) => Ok(Some(Literal::float(v))),
+                    _ => Err(invalid_err("float")),
+                },
+                RawLiteralEnum::Double(v) => match ty {
+                    Type::Primitive(PrimitiveType::Double) => Ok(Some(Literal::double(v))),
+                    _ => Err(invalid_err("double")),
+                },
+                RawLiteralEnum::String(v) => match ty {
+                    Type::Primitive(PrimitiveType::String) => Ok(Some(Literal::string(v))),
+                    _ => Err(invalid_err("string")),
+                },
+                // # TODO
+                // rust avro don't support deserialize any bytes representation now.
+                RawLiteralEnum::Bytes(_) => Err(invalid_err_with_reason(
```

Comment:
Sorry, I don't quite get your points, I think avro has byte type: https://docs.rs/apache-avro/0.16.0/apache_avro/types/enum.Value.html#variant.Bytes

Reply:
https://github.com/apache/avro/blob/2b1955947ab446ad437f152ec2f3310ea399a015/lang/rust/avro/src/de.rs#L279 But it can't support to deserialize bytes. 

And it also can't support to serialize fixed type now. https://issues.apache.org/jira/browse/AVRO-3892?filter=-2

I will try to send a PR to fix them later.

Reply:
OK, I see. I'll create an issue to track this.